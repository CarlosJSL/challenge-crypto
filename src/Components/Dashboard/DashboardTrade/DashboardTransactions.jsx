import React from 'react';
import './DashboardTransactions.css';

export default (props) => (
    <table className="ui very basic collapsing celled table">
            <thead>
                <tr>
                    <th>Trocou</th>
                    <th>Por</th>
                </tr>
            </thead>
            <tbody>
                { 
                    props.transactions.map((transaction,index) => { 
                        return (
                            <tr key ={index}>
                            <td>
                                <h4 className="ui image header">
                                    <div className="content">
                                        {transaction.sell.totalValue} {transaction.sell.crypto}
                                        <div className="sub header">{transaction.date}</div>
                                    </div>
                                </h4>
                            </td>
                            <td>
                                <h4 className="ui image header">
                                    <div className="content">
                                        {transaction.buy.totalValue} {transaction.buy.crypto}
                                        <div className="sub header">{transaction.date}</div>
                                    </div>
                                </h4>
                            </td>
                            </tr>
                        )
                    })
                }
            </tbody>
    </table>
)