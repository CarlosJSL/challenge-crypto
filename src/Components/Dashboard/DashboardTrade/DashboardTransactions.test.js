import React from 'react';
import DashboardTransactions from './DashboardTransactions';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('DashboardTransactions component', () => {
    test('should render', () => {

      const wrapper = mount(<DashboardTransactions transactions = {[]}/> );
      expect(wrapper).toMatchSnapshot();
    })
})