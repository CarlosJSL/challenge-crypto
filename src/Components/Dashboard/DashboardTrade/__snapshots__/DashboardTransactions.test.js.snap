// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTransactions component should render 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <_default
    transactions={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "transactions": Array [],
    },
    "ref": null,
    "rendered": Object {
      "instance": <table
        class="ui very basic collapsing celled table"
      >
        <thead>
          <tr>
            <th>
              Trocou
            </th>
            <th>
              Por
            </th>
          </tr>
        </thead>
        <tbody />
      </table>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                Trocou
              </th>
              <th>
                Por
              </th>
            </tr>
          </thead>,
          <tbody />,
        ],
        "className": "ui very basic collapsing celled table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <thead>
            <tr>
              <th>
                Trocou
              </th>
              <th>
                Por
              </th>
            </tr>
          </thead>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                Trocou
              </th>
              <th>
                Por
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <tr>
                <th>
                  Trocou
                </th>
                <th>
                  Por
                </th>
              </tr>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Trocou
                  </th>,
                  <th>
                    Por
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <th>
                    Trocou
                  </th>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Trocou",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Trocou",
                  ],
                  "type": "th",
                },
                Object {
                  "instance": <th>
                    Por
                  </th>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Por",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Por",
                  ],
                  "type": "th",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "thead",
        },
        Object {
          "instance": <tbody />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [
            Array [],
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "transactions": Array [],
      },
      "ref": null,
      "rendered": Object {
        "instance": <table
          class="ui very basic collapsing celled table"
        >
          <thead>
            <tr>
              <th>
                Trocou
              </th>
              <th>
                Por
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Trocou
                </th>
                <th>
                  Por
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "className": "ui very basic collapsing celled table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <thead>
              <tr>
                <th>
                  Trocou
                </th>
                <th>
                  Por
                </th>
              </tr>
            </thead>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Trocou
                </th>
                <th>
                  Por
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <tr>
                  <th>
                    Trocou
                  </th>
                  <th>
                    Por
                  </th>
                </tr>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Trocou
                    </th>,
                    <th>
                      Por
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <th>
                      Trocou
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Trocou",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Trocou",
                    ],
                    "type": "th",
                  },
                  Object {
                    "instance": <th>
                      Por
                    </th>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Por",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Por",
                    ],
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "thead",
          },
          Object {
            "instance": <tbody />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
